src/pages/notebook.tsx

```diff
@@ -64,6 +64,22 @@ export const notebookPlugin: JupyterFrontEndPlugin<void> = {
           });
         }
 
+        interface IKernelSpecMetadata {
+          name: string;
+          display_name?: string;
+        }
+
+        // Detect kernel from kernelspec,
+        const kernelspec = content.metadata?.kernelspec as IKernelSpecMetadata | undefined;
+        let kernelName: string | undefined;
+
+        if (kernelspec?.name) {
+          kernelName = kernelspec.name;
+          console.log(`Detected kernel from kernelspec: ${kernelName}`);
+        } else {
+          console.log('No kernelspec found in shared notebook.');
+        }
```

Comment:
What purpose does this code serve? `kernelName` is not used going forward, right?